src/software/ai/hl/stp/tactic/goalie_tactic_test.cpp

```diff
@@ -131,13 +134,14 @@ class GoalieTacticTest : public testing::Test
 
     void expectChipAction(Ball ball)
     {
-        World world         = ::TestUtil::createBlankTestingWorld();
-        world.mutableBall() = std::move(ball);
+        World world = ::TestUtil::createBlankTestingWorld();
+        world.updateBallStateWithTimestamp(ball.currentState());
```

Comment:
We use `Ball` everywhere else in `World`, why are we using `TimestampedBallState` here? Seems like we're mixing levels of abstraction?

Reply:
We give the world a new ball state to update the ball and the most recent timestamp. It calls ball.updateballstate and updates the internal most recent timestamp. I think it makes sense